### PR TITLE
Products list Added With Filter and Clear Filter

### PR DESCRIPTION
**Pull request by m26unkwn**
Things to Review
1. Sate-reducer
2. product-list Everything inside products list

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -8,12 +8,16 @@ import { useEffect } from "react";
...
+import { getDataFromServer } from "./services/get-data-server";
+
 function App() {
   const { dispatch } = useData();
   const [products] = useAxios("/api/products", "get");
 
   useEffect(() => {
-    dispatch({ type: "ADD_TO_PRODUCTS", payload: products });
+    getDataFromServer("/api/products", "get", dispatch, "ADD_TO_PRODUCTS");
+    //dispatch({ type: "ADD_TO_PRODUCTS", payload: products });
```
Comment (pankaj-prog) [inline src/App.js]: remove commented code

File: src/App.js [hunk 2]
```diff
@@ -8,12 +8,16 @@ import { useEffect } from "react";
...
+import { ProductList } from "./screens/ProductList";
+import { getDataFromServer } from "./services/get-data-server";
+
 function App() {
   const { dispatch } = useData();
   const [products] = useAxios("/api/products", "get");
 
   useEffect(() => {
-    dispatch({ type: "ADD_TO_PRODUCTS", payload: products });
+    getDataFromServer("/api/products", "get", dispatch, "ADD_TO_PRODUCTS");
```
Comment (pankaj-prog) [inline src/App.js:16]: What's the need of fetching products data twice? you should remove one of them.

File: src/Context/state-reducer.js [hunk 3]
```diff
@@ -2,6 +2,45 @@ export function stateReducer(state, action) {
...
+            data: found
+              ? state.selectedFilters[action.payload.filterParam].data.filter(
+                  (value) => value !== action.payload.data
+                )
+              : state.selectedFilters[action.payload.filterParam]
+              ? [
+                  ...state.selectedFilters[action.payload.filterParam].data,
+                  action.payload.data,
+                ]
+              : [action.payload.data],
```
Comment (pankaj-prog) [inline src/Context/state-reducer.js:38]: You can make this declarative by breaking task and making function to do the following task

File: src/components/productList/Products/index.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import React from "react";
+
```
Comment (pankaj-prog) [inline src/components/productList/Products/index.js]: you can change the name of this file to allproducts  from index.js

**Review by pankaj-prog — APPROVED** [review]

File: src/App.js (hunk 2 above)
Comment (m26unkwn) [reply src/App.js:16]: One is removed i guess.

File: src/components/Navbar/SidebarNav.js [hunk 5]
```diff
@@ -1,20 +1,10 @@
-import React, { useEffect } from "react";
+import React from "react";
```
Comment (PuneetP16) [inline src/components/Navbar/SidebarNav.js]: no need for writing import {React} at individual components.

File: src/components/productList/Filter/Checkbox.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+      <div className='card-divider'></div>
+
+      <li className='filter-section-title'>{filter.name}</li>
+      {filter.data.map((check) => (
+        <li key={check.id}>
+          <Input
+            type='CHECKBOX_CHANGE'
+            filterType={filter.type}
+            inputType='checkbox'
+            label={check.label}
```
Comment (PuneetP16) [inline src/components/productList/Filter/Checkbox.js:23]: You can provide a better specific name for dispatch type as it was confusing at the first glance. 
Example:
`
 type='CHECKBOX_CHANGE'
`

`
 dispatchType='CHECKBOX_CHANGE'
`

File: src/components/productList/Filter/index.js [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+            Price Low to High
+          </label>
+        </li>
+        <div class='card-divider'></div>
+        <li class='filter-section-title'>CATEGORIES</li>
+        <li>
+          <label class='form-label'>
+            <input type='checkbox' />
+            Hat
+          </label>
```
Comment (PuneetP16) [inline src/components/productList/Filter/index.js]: remove commented code

**Review by PuneetP16 — APPROVED** [review]

File: src/components/Navbar/SidebarNav.js (hunk 5 above)
Comment (saurabhchirde) [inline src/components/Navbar/SidebarNav.js]: You can omit importing react as you are using react v17.

File: src/components/home/HomeProducts/index.js [hunk 8]
```diff
@@ -1,19 +1,21 @@
 import React from "react";
```
Comment (saurabhchirde) [inline src/components/home/HomeProducts/index.js:1]: here as well.

File: src/components/productList/Filter/index.js [hunk 9]
```diff
@@ -0,0 +1,112 @@
+import React from "react";
```
Comment (saurabhchirde) [inline src/components/productList/Filter/index.js:1]: here as well.

**Review by saurabhchirde — APPROVED** [review]
Your code is clean and easily readable, 
Looks good to merge.

File: src/components/productList/Filter/index.js (hunk 9 above)
Comment (m26unkwn) [reply src/components/productList/Filter/index.js:1]: Done with your suggestion Saurabh

**Pull request merged**